### PR TITLE
#2290-Improve questionset

### PR DESCRIPTION
**Pull request by ssethupathy**

### CODE REVIEW

File: sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/QuestionCardLayout.js [hunk 1]
```diff
@@ -41,10 +41,13 @@ function QuestionCardLayout(props) {
                     showAddAnswer={showAddAnswerButton}
                 />
                 {typeof deleteCard === 'function' && (
-                    <AddIcon
+                    <button
                         className="deleteButton"
                         onClick={deleteCard}
-                    />
+                        aria-label="Delete button"
```
Comment (emmachughes) [inline sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/QuestionCardLayout.js:47]: `aria-label` should describe what the button does, not what the button is. This must also be translatable.

File: sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/Image/ImageLayout.js [hunk 2]
```diff
@@ -21,14 +21,15 @@ function ImageLayout(props) {
...
-        icon = <ImageIcon />;
+        icon = <ImageIcon style={{cursor: 'pointer'}} />;
         if ( readOnly === false ) {
             icon = (
                 <DropZone
                     onDropAccepted={onDrop}
                     multiple={false}
                     className="imageDropzone"
                     disabled={readOnly}
+                    aria-label="Add image"
```
Comment (emmachughes) [inline sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/Image/ImageLayout.js:32]: The button is the `<input type="file">` created by the DropZone component. The `aria-label` must be placed on that.

image

When using keyboard navigation, the button also has no indication it is focused.

File: sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/Answer/AnswerLayout.js [hunk 3]
```diff
@@ -76,10 +77,13 @@ const AnswerLayout = props => {
...
 
             {canDelete === true && typeof deleteAnswer === 'function' && (
-                <Icon
-                    className="deleteButton"
-                    onClick={deleteAnswer}
-                >add</Icon>
+                 <button
+                 className="deleteButton"
+                 onClick={deleteAnswer}
+                 aria-label="Delete button"
```
Comment (emmachughes) [inline sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/Answer/AnswerLayout.js:83]: Not translatable, doesn't describe what the button does.

File: sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/AddAnswer/index.js [hunk 4]
```diff
@@ -29,16 +29,13 @@ const AddAnswer = props => {
...
-                    style={styles.button}
-                    aria-label={label}
-                >
-                    {icon}
-                </Button>
-            </div>
+        <div className="addAnswerContainer" onClick={onClick} style={styles.dividerContainer}>
+            <Button
+                style={styles.button}
+                aria-label="Add answers"
```
Comment (emmachughes) [inline sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/AddAnswer/index.js:35]: Must be translatable, and only one answer is added.

**Review by emmachughes — CHANGES_REQUESTED** [review]

File: sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/QuestionCard.scss [hunk 5]
```diff
@@ -198,17 +198,11 @@
   right: 0;
   transform: rotate(45deg);
   opacity: .5;
+  border: none;
+  background: none;
+  padding-right: 0px;
```
Comment (chrieinv) [inline sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/QuestionCard.scss:203]: Don't specify length unit for 0 values

File: sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/QuestionCardLayout.js [hunk 6]
```diff
@@ -24,6 +24,7 @@ function QuestionCardLayout(props) {
         showAddAnswerButton = true,
     } = props;
 
+    const intl = useIntl()
```
Comment (chrieinv) [inline sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/QuestionCardLayout.js:27]: Prefer to deconstruct: `const { formatMessage }  = useIntl();`

File: sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/AddAnswer/index.js [hunk 7]
```diff
@@ -28,17 +29,16 @@ const AddAnswer = props => {
         },
     };
 
+    const intl = useIntl();
```
Comment (chrieinv) [inline sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/AddAnswer/index.js:32]: Deconstruct

File: sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/AddAnswer/index.js [hunk 8]
```diff
@@ -28,17 +29,16 @@ const AddAnswer = props => {
...
-            <div style={styles.dividerContainer}>
-                <Button
-                    onClick={onClick}
-                    style={styles.button}
-                    aria-label={label}
-                >
-                    {icon}
-                </Button>
-            </div>
+        <div className="addAnswerContainer" onClick={onClick} style={styles.dividerContainer}>
```
Comment (chrieinv) [inline sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/AddAnswer/index.js:35]: Move the CSS styles to the CSS class used, button styles can also be moved. The `styles.divider` seems unused.

File: sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/AddCard/index.js [hunk 9]
```diff
@@ -17,18 +19,23 @@ const AddCard = props => {
             fontWeight: '400 !important',
         }
     };
+    const useStyles = makeStyles((theme) => ({
+        largeButton: {
+            width: '100%',
+        },
+    }));
+    const classes = useStyles();
+    const intl = useIntl();
```
Comment (chrieinv) [inline sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/AddCard/index.js:28]: Deconstruct

File: sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/AddCard/index.js [hunk 10]
```diff
@@ -17,18 +19,23 @@ const AddCard = props => {
             fontWeight: '400 !important',
         }
     };
+    const useStyles = makeStyles((theme) => ({
```
Comment (chrieinv) [inline sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/AddCard/index.js:22]: Avoid using several different ways of styling. Use either the current or the new, not both. So either move `addNewQuestionContainer` to `useStyles` or move `largeButton` to `styles`.

File: sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/Answer/AnswerLayout.js [hunk 11]
```diff
@@ -33,6 +34,7 @@ const AnswerLayout = props => {
     } = props;
     const { editorLanguage } = useEditorSetupContext();
     let inputType;
+    const intl = useIntl();
```
Comment (chrieinv) [inline sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/Answer/AnswerLayout.js:37]: Deconstruct

File: sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/Image/ImageLayout.js [hunk 12]
```diff
@@ -19,16 +21,28 @@ function ImageLayout(props) {
         uploading,
     } = props;
 
+    const intl = useIntl();
```
Comment (chrieinv) [inline sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/Image/ImageLayout.js:24]: Deconstruct

File: sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/Image/ImageLayout.js [hunk 13]
```diff
@@ -1,10 +1,12 @@
 import React from 'react';
+import { useState } from 'react';
```
Comment (chrieinv) [inline sourcecode/apis/contentauthor/resources/assets/react/components/QuestionContentType/components/QuestionCard/components/Image/ImageLayout.js:2]: Join with line 1

File: sourcecode/apis/contentauthor/resources/assets/react/components/RichEditor/RichEditor.scss [hunk 14]
```diff
@@ -49,4 +49,7 @@
 
 .richeditor-readonly {
   cursor: not-allowed;
-}
\ No newline at end of file
+}
+
```
Comment (chrieinv) [inline sourcecode/apis/contentauthor/resources/assets/react/components/RichEditor/RichEditor.scss:53]: There should be only one empty line at the end of files

**Review by chrieinv — CHANGES_REQUESTED** [review]

**Review by chrieinv — CHANGES_REQUESTED** [review]

**Review by chrieinv — APPROVED** [review]

**Review by emmachughes — APPROVED** [review]

**Pull request merged**